src/main.cpp

```diff
@@ -1581,6 +1591,8 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         BOOST_FOREACH(const uint256& hashTx, vHashTxToUncache)
             pcoinsTip->Uncache(hashTx);
     }
+    // After we've (potentially) uncached entries, ensure our coins cache is still within its size limits
+    FlushStateToDisk(state, FLUSH_STATE_PERIODIC);
```

Comment:
FlushStateToDisk shouldn't effect anything that mempool (or anything, actually) cares about except for memory usage, I believe.